src/Database/PostgreSQL/Consumers/Components.hs

```diff
@@ -186,14 +213,23 @@ spawnDispatcher
   -> MVar ()
   -> TVar (M.Map ThreadId idx)
   -> TVar Int
+  -> Maybe (TMVar Bool)
   -> m ThreadId
 spawnDispatcher ConsumerConfig{..} useSkipLocked cs cid semaphore
-  runningJobsInfo runningJobs =
+  runningJobsInfo runningJobs mIdleSignal =
   forkP "dispatcher" . forever $ do
     void $ takeMVar semaphore
-    loop 1
+    ifM (loop 1) (setIdle False) (setIdle True)
```

Comment:
I'd write this as
```
batchSizeNonzero <- loop 1
if batchSizeNonzero then .. else
```
because `loop` loops for a while and then returns, so writing it like this IMO makes it more obvious.

Reply:
done